mws/mws.py

```diff
@@ -76,18 +85,31 @@ def calc_request_description(params):
     """
     description_items = []
     for item in sorted(params.keys()):
-        encoded_val = quote(str(params[item]), safe='-_.~')
+        encoded_val = params[item]
         description_items.append('{}={}'.format(item, encoded_val))
     return '&'.join(description_items)
 
 
 def remove_empty(dict_):
-    """
-    Returns dict_ with all empty values removed.
-    """
     return {k: v for k, v in dict_.items() if v}
 
 
+def clean_extra_data(extra_data):
+    extra_data_enc = dict()
+    for key, value in extra_data.items():
+        if isinstance(value, (datetime.datetime, datetime.date)):
+            value = str(value.isoformat())
```

Comment:
The `str()` conversion isn't necessary: `isoformat` outputs a string in all cases.